app/actions/chatActions.ts

```diff
@@ -0,0 +1,45 @@
+'use server';
+
+import { GoogleGenerativeAI } from "@google/generative-ai";
+import { z } from "zod";
+
+interface FormState {
+  response: string;
+  error: string | null;
+}
+
+const ChatSchema = z.object({
+  userQuestion: z.string().min(1, { message: "メッセージを入力してください。" }),
+  documentContent: z.string().min(1, { message: "コンテキストとなるドキュメントがありません。"}),
+});
+
+const genAI = new GoogleGenerativeAI(process.env.API_KEY!);
+
+export async function askAI(params: unknown): Promise<FormState> {
```

Comment:
```suggestion
type ChatParams = z.input<typeof ChatSchema>;
export async function askAI(params: ChatParams): Promise<FormState> {
```

TypeScriptの型指定はコードを書く際の間違いをチェックするためのもの、validationは実際の値のチェックのためで、目的が異なり両方必要です

```ts
type ChatParams = z.input<typeof ChatSchema>;
```
と書くことで
```ts
interface ChatParams {
  userQuestion: string;
  documentContent: string;
}
```
と同じものがChatSchemaの定義から自動で作られます